test/Dialect/BaseModelica/Transforms/CallCSE/basic-cse.mlir

```diff
@@ -0,0 +1,61 @@
+// RUN: modelica-opt %s --call-cse | FileCheck %s
+
+// CHECK-LABEL: @Test
+module @Test {
+	bmodelica.function @foo {
+		bmodelica.variable @x : !bmodelica.variable<f64, input>
+		bmodelica.variable @y : !bmodelica.variable<f64, output>
+
+		bmodelica.algorithm {
+			%0 = bmodelica.variable_get @x : f64
+			bmodelica.variable_set @y, %0 : f64
+		}
+	}
+
+    // CHECK: bmodelica.model
+	bmodelica.model @M {
+		// CHECK: bmodelica.variable @[[CSE:_cse0_0]]
```

Comment:
Check the type of the variable. Also, don't hardcode the name. You can check also for the already existing variables `x` and `y` to ensure that the CSE variable gets created.